pkg/logql/syntax/ast.go

```diff
@@ -1562,3 +1565,38 @@ func MatcherGroups(expr Expr) []MatcherRange {
 		return nil
 	}
 }
+
+type VectorExpr struct {
+	Val float64
+	implicit
+}
+
+func NewVectorExpr(scalar string) *VectorExpr {
+	n, err := strconv.ParseFloat(scalar, 64)
+	if err != nil {
+		panic(logqlmodel.NewParseError(fmt.Sprintf("unable to parse vectorExpr as a float: %s", err.Error()), 0, 0))
```

Comment:
Do we have to panic here?

Reply:
Other Exprs are panic too.

![image](https://user-images.githubusercontent.com/9583245/187618233-c63591c6-52b7-4ee1-9e50-f0f50a81f2bb.png)

Reply:
Hhhmm, that's not great.
I think that's a problem, and we should fix it, so let's not make the problem worse here IMHO.
I think we should remove the panic. Maybe we can find a way to make this fail gracefully without panicking the whole querier? Perhaps setting the vector value as some crazy number like `MaxFloat64` if the parsing fails, and if this value is present when querying, return an error that bubbles up to the user to give a helpful error message.

Reply:
https://pkg.go.dev/bufio#Scanner.Scan

Does VectorExpr refer to Scanner to provide an Err() function?

![image](https://user-images.githubusercontent.com/9583245/187621346-c073b6a7-ccf6-43fe-80f9-981f3e927004.png)

Reply:
Probably changed to look like this.
![image](https://user-images.githubusercontent.com/9583245/187622038-9fd187d8-13b3-4406-afc1-6ad1481b1164.png)